data_steward/cdr_cleaner/cleaning_rules/deid/year_of_birth_records_suppression.py

```diff
@@ -0,0 +1,295 @@
+
+    def setup_rule(self, client):
+        self._get_time_columns(client)
+        self._get_observation_concept_id_columns(client)
+
+    def _get_observation_concept_id_columns(self, client):
+        tables_columns_query_template = JINJA_ENV.from_string("""
+        SELECT 
+           column_name
+         FROM
+           `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS`
+         WHERE
+            (lower(data_type) in ("int64") and REGEXP_CONTAINS(column_name, r'(?i)(concept_id)'))
+            -- tables we are not interested in cleaning --
+           and lower(table_name) = 'observation'
+        """)
+        tables_columns_query = tables_columns_query_template.render(
+            project=self.project_id, dataset=self.dataset_id)
+
+        try:
+            response = client.query(tables_columns_query,
+                                    job_id_prefix='ct_yob_setup_')
+        except GoogleCloudError as exc:
+            raise exc
+        else:
+            response_list = list(response.result())
+
+            # move all the columns to key/value dictionary list for JINJA templating
+            new_response_list = []
+            for item in response_list:
+                new_response_list.append(item[0])
+
+        self.observation_concept_id_columns = new_response_list
+
+    def _get_time_columns(self, client):
+        tables_columns_query_template = JINJA_ENV.from_string("""
+        CREATE OR REPLACE TABLE `{{project}}.{{sandbox_dataset}}.{{lookup_table}}` AS (
+            SELECT 
+               table_name, column_name
+             FROM
+               `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS`
+             WHERE
+                (lower(data_type) in ("date", "datetime", "time", "timestamp") and not REGEXP_CONTAINS(column_name, r'(?i)(partitiontime)'))
+                -- tables we are not interested in cleaning --
+               and not REGEXP_CONTAINS(table_name, '(?i)(_ext)|(person)|(activity_summary)|(steps_intraday)|(heart_rate)')
+
+               -- make sure table has a person_id column --
+                and table_name in (
+                    select table_name 
+                    from `{{project}}.{{dataset}}.INFORMATION_SCHEMA.COLUMNS` 
+                    where lower(column_name) = 'person_id'
+                )
+            ORDER BY 1,2
+        )
+        """)
+        tables_columns_query = tables_columns_query_template.render(
+            project=self.project_id,
+            dataset=self.dataset_id,
+            sandbox_dataset=self.sandbox_dataset_id,
+            lookup_table=self.sandbox_table_for(LOOKUP_TABLE))
+
+        try:
+            client.query(tables_columns_query, job_id_prefix='ct_yob_setup_')
+        except GoogleCloudError as exc:
+            raise exc
+        else:
+            # Add 2 seconds to ensure the lookup table job is finished creating before reading from the table.
+            time.sleep(2)
```

Comment:
I think we can get rid of a timed wait if we wait for the previous query to finish executing.  I am attaching a patch file.
[patch_using_result.txt](https://github.com/all-of-us/curation/files/8235146/patch_using_result.txt)

Reply:
Thank you for this - I made one tweak to the patch to just execute:
`response.result()`

rather than:
`response = response.result()`

It worked as desired in local testing.